Prog/NCHousing_needs_units_targets.sas

```diff
@@ -0,0 +1,927 @@
+	    month_mortgage= (4.1 / 12) / 100; 
+	    monthly_PI = loan * month_mortgage * ((1+month_mortgage)**360)/(((1+month_mortgage)**360)-1);
+
+	    ****
+	    Calculate PMI and taxes/insurance to add to Monthly_PI to find total monthly payment
+	    ******;
+	    
+	    PMI = (.007 * loan ) / 12; **typical annual PMI is .007 of loan amount;
+	    tax_ins = .25 * monthly_PI; **taxes assumed to be 25% of monthly PI; 
+	    total_month = monthly_PI + PMI + tax_ins; **Sum of monthly payment components;
+		
+			/*create owner cost level categories*/ 
+			ownlevel=.;
+				if 0 <=total_month<1200 then ownlevel=1;
+				if 1200 <=total_month<1800 then ownlevel=2;
+				if 1800 <=total_month<2500 then ownlevel=3;
+				if 2500 <=total_month<3200 then ownlevel=4;
+				if 3200 <=total_month<4200 then ownlevel=5;
+				if total_month >= 4200 then ownlevel=6;
+			
+			/*create  categories now used in targets for renter/owner costs combined*/ 
+				allcostlevel=.;
+				if total_month<800 then allcostlevel=1;
+				if 800 <=total_month<1300 then allcostlevel=2;
+				if 1300 <=total_month<1800 then allcostlevel=3;
+				if 1800 <=total_month<2500 then allcostlevel=4;
+				if 2500 <=total_month<3500 then allcostlevel=5;
+				if total_month >= 3500 then allcostlevel=6; 
+
+
+	  end;
+
+
+	  paycategory=4; *add vacant as a category to paycategory; 
+
+		if BUILTYR2 in ( 00, 9999999, .n , . ) then structureyear=.;
+		else do; 
+		    if BUILTYR2  in (07, 08, 09, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22) then structureyear=1;
+			else if BUILTYR2  in (04, 05, 06) then structureyear=2;
+            else if BUILTYR2 in (01, 02, 03)  then structureyear=3;
+		end;
+		
+  *add structure of housing variable;
+    if UNITSSTR =00 then structure=5;
+	if UNITSSTR in (01, 02) then structure=4;
+	if UNITSSTR in (03, 04) then structure=1;
+	if UNITSSTR in (05, 06, 07) then structure=2;
+	if UNITSSTR in (08, 09, 10) then structure=3;
+
+		label rentlevel = 'Rent Level Categories based on Current Gross Rent'
+		 		  allcostlevel='Housing Cost Categories (tenure combined) based on Current Rent or First-time Buyer Mtg'
+				  ownlevel = 'Owner Cost Categories based on First-Time HomeBuyer Costs'
+				  paycategory = "Whether Occupant pays too much, the right amount or too little" 
+				  structureyear = 'Age of structure'
+				  structure = 'Housing structure type'
+				;
+	format ownlevel ocost. rentlevel rcost. vacancy_r VACANCY_F. allcostlevel acost. ; 
+
+	*output other vacant - seasonal separately ;
+	if vacancy in (1, 2, 3) then output Housing_needs_vacant_&year.;
+	else if vacancy in (4, 7, 9) then output Other_vacant_&year.; 
+	run;
+
+%mend single_year; 
+
+%single_year(2013);
+%single_year(2014);
+%single_year(2015); 
+%single_year(2016);
+%single_year(2017);
+
+/*merge single year data and reweight
+
+revised to match Steven's files in https://urbanorg.box.com/s/nm4wb0arxsxv8aml98b6y062tudzp2i6
+*/
+data projectedHH;
+set NChsg.projection_for_calibration;
```

Comment:
did you need to rerun this based on the adjustments that were made today?

Reply:
yes it is updated.